azurerm/internal/services/network/network_connection_monitor_resource.go

```diff
@@ -200,10 +200,30 @@ func resourceNetworkConnectionMonitor() *schema.Resource {
 							},
 						},
 
+						"resource_id": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Computed: true,
```

Comment:
```suggestion
							Type:          schema.TypeString,
							Optional:      true,
							Computed:      true,
							ConflictsWith: []string{"virtual_machine_id"},
```

Reply:
After tested, ConflictsWith doesn't work in TypeSet and it would throw below error message. So I have to add logic to judge it.

```
Error: Internal validation of the provider failed! This is always a bug
with the provider itself, and not a user issue. Please report
this bug:

1 error occurred:
        * resource azurerm_network_connection_monitor: ConflictsWith: resource_id references unknown attribute (virtual_machine_id) at part (virtual_machine_id)
```

Reply:
Ah ok, sorry I missed that. In that case let's prioritize the new property, and read the old property second.

Reply:
Sorry. I am not sure what you mean. Could you give an example?

Reply:
Sorry it's out of context here, but never mind actually, I see you added an apply time check at line 625 which is fine 👍